controllers/core/function_controller.go

```diff
@@ -581,6 +582,14 @@ func (r *FunctionReconciler) createServingSpec(fn *openfunction.Function) openfu
 	return spec
 }
 
+func getServingImage(fn *openfunction.Function) string {
+	if fn.Status.Output == nil || fn.Status.Output.Digest == "" {
+		return fn.Spec.Image
+	}
+
+	return fn.Spec.Image + "@" + fn.Status.Output.Digest
```

Comment:
What happens if `fn.Spec.Image` already contains digest

Reply:
Good point, we should remove existing digest

Reply:
There will be an `ImagePullBackOff` error.

We can state in the document that setting sha256 in the image is not supported.